Drop CentOS 7 and 8 support

smortex opened the pull request:
These versions or CentOS have reached end-of-life and the images used
for CI are not available anymore.  As we cannot ensure these legacy
systems are still working with the module, remove them from
metadata.json.

~~Also include:~~
  * ~~#45~~

smortex committed:
Drop CentOS 7 and 8 support

These versions or CentOS have reached end-of-life and the images used
for CI are not available anymore.  As we cannot ensure these legacy
systems are still working with the module, remove them from
metadata.json.

faxm0dem commented:
I'd be delighted to merge "Drop Centos 7 and 8" support, but I'd also be merging 6 other unrelated commits.

smortex commented:
Yes they are all part of #45, CI show that CentOS 7 & 8 are broken so I removed them in a separate PR because this change is backwards incompatible.  I rebased this branch specific changes on top of master.

smortex commented:
> This is a scheduled Ubuntu 20.04 retirement. Ubuntu 20.04 LTS runner will be removed on 2025-04-15. For more details, see https://github.com/actions/runner-images/issues/11101

Dang!

faxm0dem reviewed (commented):
We might as well drop the whole CentOS section, don't you think ?

smortex commented:
> We might as well drop the whole CentOS section, don't you think ?

I think that CentOS 9 / 10 (aka CentOS Stream) can still be tested with the vox tooling.  Those would be added with the corresponding versions of RedHat / AlmaLinux / Rocky / etc.

I have not tried it, and decommissioned the last rpm-based Operating System from our fleet last year, so it is not something I will spend time on.  Also, adding OS is a minor change so will not result in a major version bump so I am fine with keeeping this as it is (so that CentOS is still listed as supported on the Forge, but is untested and so if people have trouble they know they can send fix PRs).  I am also fine for removing it completely if you prefer the module not listing CentOS as supported at all.

Also, I belive that with the updated tooling (see https://github.com/ccin2p3/modulesync_config/pull/7) outdated OS are skipped, and so these failures would not appear anymore, with CentOS 7 & 8 still listed as supported.  Not ideal since because they are untested they can silently break, but still a valid option.

faxm0dem reviewed (approved):
LGTM, thanks !
